dev/lib/edit-map.js

```diff
@@ -109,13 +109,16 @@ export class EditMap {
       events.length = this.map[index][0]
     }
 
-    vecs.push([...events])
+    vecs.push(events.slice())
     events.length = 0
 
     let slice = vecs.pop()
 
     while (slice) {
-      events.push(...slice)
+      // NOTE: do not use spread operator here, it will cause a stack overflow.
+      for (let i = 0; i < slice.length; i++) {
```

Comment:
Adding a loop could could significant negative performance impact.
Could concat be used instead?

See:
https://github.com/micromark/micromark/issues/187
https://github.com/micromark/micromark/commit/acc135ed366fdeac900caf69b66a502ee993724e

Reply:
I try to use `Array.contact` to replace `for loop`, but the `event` is a parameter, I just can modify it in-place, do you have any recommend, thx.